src/unicorn/mod.rs

```diff
@@ -24,11 +24,12 @@ pub mod qubot;
 pub mod sat_solver;
 pub mod smt_solver;
 pub mod unroller;
+pub mod horizon;
 
 pub type Nid = u64;
 pub type NodeRef = Rc<RefCell<Node>>;
 
-#[derive(Debug)]
+#[derive(Clone,Debug)]
```

Comment:
I understand that we need to clone `Model` for now. But from what I can tell we are not cloning `Node` objects. I might very well be missing something. Could you double-check whether deriving `Clone` on `Node` is still necessary here?

Reply:
True, this is a leftover that I will fix - thanks!